File: clock/Clock.fs
module Clock

type Clock = | Minutes of int

let (%%) x y = y % x

let create hours minutes = 
    let withinDay m = if m < 0 then 1440 + m else m
    minutes + (hours * 60) |> (%%) 1440 |> withinDay |> Minutes

let add minutes clock = 
    let (Minutes existing) = clock
    create 0 (existing + minutes)

let subtract minutes clock = 
    let (Minutes existing) = clock
    create 0 (existing - minutes)

let display clock = 
    let (Minutes existing) = clock
    sprintf "%02i:%02i" 
        ((float existing) / 60.0 |> floor |> int |> (%%) 24) 
        (existing % 60)

File: csharp/clock/Clock.cs
using System;

public struct Clock
{
    private readonly int totalMinutes;

    const int minutesInDay = 24 * 60;

    public Clock(int hours, int minutes)
    {
        totalMinutes = (hours * 60 + minutes) % minutesInDay;
        if (totalMinutes < 0) totalMinutes += minutesInDay;
    }

    public int Hours => totalMinutes / 60;

    public int Minutes => totalMinutes % 60;

    public Clock Add(int minutesToAdd) => new Clock(0, totalMinutes + minutesToAdd);

    public Clock Subtract(int minutesToSubtract) => new Clock(0, totalMinutes - minutesToSubtract);

    public override string ToString() => $"{Hours:00}:{Minutes:00}";
}